nifi-extension-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/GenerateFlowFile.java

```diff
@@ -71,12 +71,36 @@ public class GenerateFlowFile extends AbstractProcessor {
     public static final String DATA_FORMAT_BINARY = "Binary";
     public static final String DATA_FORMAT_TEXT = "Text";
 
+    public static final PropertyDescriptor VARIABLE_SIZE = new PropertyDescriptor.Builder()
+            .name("Variable File Size")
+            .description("The size of the FlowFile will be randomized between a minimum and maximum size value. 'Unique FlowFiles' must be set to true in order to achieve varying sizes.")
+            .required(true)
```

Comment:
why make all these required?

Reply:
I was keeping in style with the existing File Size attribute. I can make not required if you prefer - and perhaps add some custom validation to be sure min/max size is specified when Variable File Size is true.

Reply:
@joewitt I made the min/max file sizes non-required. It makes more work for the user to have to enter values. However, I'm inclined to think the default values would not be appropriate in a majority of the cases anyhow.